### DISCUSSION TITLE
Naming question: BlendPercentage isn't really a percentage?

### DISCUSSION THREAD

**Discussion by atruskie** [SixLabors/ImageSharp#1448] [Ideas] [answered]
I have noticed that `GraphicsOptions.BlendPercentage` is not really a percentage (a number in the range `[0,100]`) .

https://github.com/SixLabors/ImageSharp/blob/78a584e8482b052d7a9885682299e2f37518d83d/src/ImageSharp/GraphicsOptions.cs#L56-L72

This is quite a nit-picky ask: but is it worth changing the name?

I wouldn't bring it up except that I keep having to refer to the XML document to understand the range of acceptable values. There is a _small_ cognitive overload each time when I realize the implied range [0,100] is not equal to the actual range [0.0, 1.0].

Alternate names:

- `BlendRatio`
- `BlendFraction`
- `BlendQuotient` (most correct, too technical?)
- `BlendAmount` (no context, no implied range)

**Comment by antonfirsov** [comment, marked as answer]
Good point! I also dislike this name just didn't care about it too much, would vote for `BlendRatio` ... if we could still change it.

Unfortunately, as we promised API stability with our 1.0 release of the main library, and I have a very strong opinion that we should keep our promises, and keep breaking changes at an absolute minimum, so I'm afraid there is no way back :(

Note that similar suggestions are more than welcome in Drawing, since it's still in beta.

**Reply by JimBobSquarePants** [reply]
Yeah. Not much we can do about that now.

**Reply by JimBobSquarePants** [reply]
I wonder... 

If we added a better named property that the old property will proxy to, marked the old property as Obsolete and used EditorBrowsable to also hide the old property from intellisense would that work?

**Reply by atruskie** [reply]
💯

I had the same idea but didn't know about EditorBrowsable and thought the intellisense noise would be too much

**Reply by antonfirsov** [reply]
This is a good idea! Note that BCL is already doing similar change (even without using `[ObsoleteAttribute]` .. yet):

https://github.com/dotnet/runtime/issues/43901

**Reply by antonfirsov** [reply]
related: #1454
